core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -557,7 +553,16 @@ trait EntityComponent {
                     on ea.value_entity_ref = referenced_entity.old_id
                 join ENTITY e on e.id = ea.owner_id where ea.deleted = false and e.deleted = false and e.workspace_id = $clonedWorkspaceId
           """
+        addEntitiesToWhereIfNeeded(entityRefs, baseInsert).head
+      }
 
+      private def addEntitiesToWhereIfNeeded(entityRefs: Set[AttributeEntityReference], baseInsert: SQLActionBuilder) = {
+        if (entityRefs.nonEmpty) {
+          val entityTypeNameTuples = reduceSqlActionsWithDelim(entityRefs.map { ref => sql"(${ref.entityType}, ${ref.entityName})" }.toSeq)
+          concatSqlActions(baseInsert, sql" and (e.entity_type, e.name) in (", entityTypeNameTuples, sql")").as[Int]
+        } else {
+          concatSqlActions(baseInsert).as[Int]
```

Comment:
concatSqlActions is extraneous here, could be just:
```suggestion
          baseInsert.as[Int]
```